pkg/ru_fa/reader.go

```diff
@@ -0,0 +1,292 @@
+var headerUnregisteredAssociation = []string{"№", "Наименование", "Дата включения в реестр", "Цели деятельности объединения", "Сведения об источниках формирования денежных средств и (или) иного имущества объединения, в том числе сведения об иностранных источниках поступления (планируемого поступления) денежных средств и иного имущества"}
+
+// ---------- Struct Definitions ----------
+
+type NonCommercialOrganization struct {
+	Number        string
+	Name          string
+	NameTranslit  string
+	Address       string
+	INN           string
+	RegistryNo    string
+	InclusionDate string
+	ExclusionDate string
+}
+
+type MassMedia struct {
+	Number        string
+	Name          string
+	NameTranslit  string
+	InclusionDate string
+	ExclusionDate string
+}
+
+type MediaIndividual struct {
+	Number        string
+	Name          string
+	NameTranslit  string
+	Activity      string
+	InclusionDate string
+	ExclusionDate string
+}
+
+// In this table, the second column is Name and the third is Activity.
+type ForeignAgentIndividual struct {
+	Number                string
+	Name                  string
+	NameTranslit          string
+	Activity              string
+	ForeignSources        string
+	PoliticalActivityInfo string
+	InclusionDate         string
+}
+
+type UnregisteredAssociation struct {
+	Number           string
+	Name             string
+	NameTranslit     string
+	InclusionDate    string
+	AssociationGoals string
+	SourceInfo       string
+}
+
+// Tables holds all table data.
+type Tables struct {
+	NonCommercialOrgs        []NonCommercialOrganization
+	MassMedias               []MassMedia
+	MediaIndividuals         []MediaIndividual
+	ForeignAgentIndividuals  []ForeignAgentIndividual
+	UnregisteredAssociations []UnregisteredAssociation
+}
+
+func Read(files download.Files) (Tables, error) {
+	for filename, file := range files {
+		switch strings.ToLower(filepath.Base(filename)) {
+		case htmlFilename:
+			tables, err := htmlFile(file)
+			if err != nil {
+				return tables, fmt.Errorf("%s: %v", htmlFilename, err)
+			}
+			return tables, nil
+		default:
+			var tables Tables
+			file.Close()
+			return tables, fmt.Errorf("error: file %s does not have a handler for processing", filename)
+		}
+	}
+
+	var tables Tables
+	return tables, fmt.Errorf("error: file %s not found", htmlFilename)
+}
+
+func htmlFile(f io.ReadCloser) (Tables, error) {
+	defer f.Close()
+	var tables Tables
+
+	// Load Moscow timezone.
+	moscowLoc, err := time.LoadLocation("Europe/Moscow")
+	if err != nil {
+		return tables, fmt.Errorf("loading timezone: %v", err)
+	}
+	now := time.Now().In(moscowLoc)
+
+	doc, err := goquery.NewDocumentFromReader(f)
+	if err != nil {
+		return tables, fmt.Errorf("parsing HTML: %v", err)
+	}
+
+	// Iterate over every table.
+	doc.Find("table").Each(func(i int, tableSel *goquery.Selection) {
```

Comment:
Is there a CSV, XML, etc format that would be more stable over time and easier to parse?

Reply:
I tried pretty hard to find CSV/XML, but I couldn't find it for some reason.